lib/response.js

```diff
@@ -143,7 +143,20 @@ internals.Response.prototype.vary = function (value) {
         this.headers.vary = value;
     }
     else if (this.headers.vary !== '*') {
-        this._header('vary', value, { append: true });
+
+        var values = this.headers.vary.split(',');
+        var foundMatch = false;
+
+        for (var i = 0, il = values.length; i < il; ++i) {
+            if (values[i] === value) {
+                foundMatch = true;
```

Comment:
I would have but there appears to be opportunities to return early in this function and it was not done. So, I was following the current flow and decided to `break`. Returning early does save an allocation and I am already doing `split` so I just changed it. Thanks!